### PR TITLE
Convert Relay to Kotlin

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/cache2/Relay.kt [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+  val isClosed: Boolean
+    get() = file == null
+
+  @Throws(IOException::class)
+  private fun writeHeader(
+    prefix: ByteString,
+    upstreamSize: Long,
+    metadataSize: Long
+  ) {
+    val header = Buffer()
```
Comment (jemaystermind) [inline okhttp/src/main/java/okhttp3/internal/cache2/Relay.kt]:
```suggestion
    val header = Buffer().apply{
      write(prefix)
      writeLong(upstreamSize)
      writeLong(metadataSize)
    }
```

File: okhttp/src/main/java/okhttp3/internal/cache2/Relay.kt [hunk 2]
```diff
@@ -13,349 +13,345 @@
...
-          return -1L;
+          commit(upstreamPos)
+          return -1L
         }
 
         // Update this source and prepare this call's result.
-        long bytesRead = Math.min(upstreamBytesRead, byteCount);
-        upstreamBuffer.copyTo(sink, 0, bytesRead);
-        sourcePos += bytesRead;
+        val bytesRead = Math.min(upstreamBytesRead, byteCount)
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/cache2/Relay.kt]: minOf

File: okhttp/src/main/java/okhttp3/internal/cache2/Relay.kt (hunk 1 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/cache2/Relay.kt]: Done.

File: okhttp/src/main/java/okhttp3/internal/cache2/Relay.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/cache2/Relay.kt]: Fixed all.

**Pull request merged**